src/main/scala/com/fulcrumgenomics/fasta/UpdateFastaContigNames.scala

```diff
@@ -24,70 +24,157 @@
 
 package com.fulcrumgenomics.fasta
 
+import java.io.BufferedWriter
+
 import com.fulcrumgenomics.FgBioDef.PathToSequenceDictionary
 import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
 import com.fulcrumgenomics.commons.CommonsDef._
 import com.fulcrumgenomics.commons.util.LazyLogging
 import com.fulcrumgenomics.sopt.{arg, clp}
 import com.fulcrumgenomics.util.{Io, ProgressLogger}
-import htsjdk.samtools.reference.{FastaSequenceIndex, ReferenceSequenceFileFactory}
+import htsjdk.samtools.reference.{FastaSequenceIndex, ReferenceSequence, ReferenceSequenceFile, ReferenceSequenceFileFactory}
 
 @clp(description =
   """
     |Updates the sequence names in a FASTA.
     |
     |The name of each sequence must match one of the names (including aliases) in the given sequence dictionary.  The
     |new name will be the primary (non-alias) name in the sequence dictionary.
+    |
+    |By default, the sort order of the contigs will be the same as the input FASTA.  Use the `--sort-by-dict` option to
+    |sort by the input sequence dictionary.  Furthermore, the sequence dictionary may contain **more** contigs than the
+    |input FASTA, and they wont be used.  Use the `--skip-missing` option to skip contigs in the input FASTA that cannot
```

Comment:
I think a little more explanation here would be good.  Assuming `--skip-missing` means to not emit the contig (not just not rename it), I'd rewrite the last two sentences as:

```
Use the `--skip-missing` option to skip contigs in the input FASTA that cannot be renamed (i.e. who are not present in the input sequence dictionary); missing contigs will not be written to the output fasta.  Finally, use the `--default-contigs` option to specify an additional FASTA which will be queried to locate contigs not present in the the input FASTA but present in the sequence dictionary.
```